File: Java/CesiumLanguageWriter/translatedSrc/cesiumlanguagewriter/advanced/CesiumArcTypeValuePropertyAdaptor.java
package cesiumlanguagewriter.advanced;


import agi.foundation.compatibility.*;
import cesiumlanguagewriter.*;
import javax.annotation.Nonnull;

/**
 *  
 Adapts a class that implements {@link ICesiumArcTypeValuePropertyWriter} to implement
 {@link ICesiumPropertyWriter} for {@link CesiumArcType} values.
 
 

 * @param <TFrom> The class that implements {@link ICesiumArcTypeValuePropertyWriter} to adapt.
 */
@SuppressWarnings( {
        "unused",
        "deprecation",
        "serial"
})
public class CesiumArcTypeValuePropertyAdaptor<TFrom extends ICesiumArcTypeValuePropertyWriter> extends CesiumWriterAdaptor<TFrom, CesiumArcType> {
    /**
    *  
    Initializes a new instance.
    
    
    
    

    * @param parent The instance to wrap.
    * @param writeValueCallback The callback to write values of type {@link CesiumArcType}.
    * @param writeDeleteValueCallback The callback to write an indication that the client should delete existing data.
    */
    public CesiumArcTypeValuePropertyAdaptor(@Nonnull TFrom parent, @Nonnull CesiumWriterAdaptorWriteCallback<TFrom, CesiumArcType> writeValueCallback,
            @Nonnull CesiumWriterAdaptorWriteDeleteCallback<TFrom> writeDeleteValueCallback) {
        super(parent, writeValueCallback, writeDeleteValueCallback);
    }
}

File: DotNet/CesiumLanguageWriter/Advanced/CesiumArcTypeValuePropertyAdaptor.cs
using JetBrains.Annotations;

namespace CesiumLanguageWriter.Advanced
{
    /// <summary>
    /// Adapts a class that implements <see cref="ICesiumArcTypeValuePropertyWriter"/> to implement
    /// <see cref="ICesiumPropertyWriter"/> for <see cref="CesiumArcType"/> values.
    /// </summary>
    /// <typeparam name="TFrom">The class that implements <see cref="ICesiumArcTypeValuePropertyWriter"/> to adapt.</typeparam>
    public class CesiumArcTypeValuePropertyAdaptor<TFrom> : CesiumWriterAdaptor<TFrom, CesiumArcType>
        where TFrom : class, ICesiumArcTypeValuePropertyWriter
    {
        /// <summary>
        /// Initializes a new instance.
        /// </summary>
        /// <param name="parent">The instance to wrap.</param>
        /// <param name="writeValueCallback">The callback to write values of type <see cref="CesiumArcType"/>.</param>
        /// <param name="writeDeleteValueCallback">The callback to write an indication that the client should delete existing data.</param>
        public CesiumArcTypeValuePropertyAdaptor([NotNull] TFrom parent,
                                                 [NotNull] CesiumWriterAdaptorWriteCallback<TFrom, CesiumArcType> writeValueCallback,
                                                 [NotNull] CesiumWriterAdaptorWriteDeleteCallback<TFrom> writeDeleteValueCallback)
            : base(parent, writeValueCallback, writeDeleteValueCallback)
        {
        }
    }
}